daft/dataframe/dataframe.py

```diff
@@ -429,6 +431,69 @@ def write_csv(
         result_df._preview = write_df._preview
         return result_df
 
+    def write_iceberg(self, table: "IcebergTable", mode: str = "append") -> "DataFrame":
+
+        if len(table.spec().fields) > 0:
+            raise ValueError("Cannot write to partitioned Iceberg tables")
+        from pyiceberg.table import _MergingSnapshotProducer
+        from pyiceberg.table.snapshots import Operation
+
+        operations = []
+        path = []
+        rows = []
+        size = []
+
+        if mode == "append":
+            operation = Operation.APPEND
+        elif mode == "overwrite":
+            operation = Operation.OVERWRITE
+        else:
+            raise ValueError(f"Only support `append` or `overwrite` mode. {mode} is unsupported")
+
+        # We perform the merge here since IcebergTable is not pickle-able
+        merge = _MergingSnapshotProducer(operation=operation, table=table)
+
+        builder = self._builder.write_iceberg(table)
```

Comment:
@Fokko can you take a look to see if this is fine?

Since we can't pickle the Iceberg Table and we still want distributed writes, we're having daft write out the parquet files and then return pyiceberg DataFiles in the distributed dataframe. Then we commit the data files with `_MergingSnapshotProducer`.

Reply:
Hey @samster25

> Since we can't pickle the Iceberg Table and we still want distributed writes, we're having daft write out the parquet files and then return pyiceberg DataFiles in the distributed dataframe.

That sounds like a good approach!

> Then we commit the data files with _MergingSnapshotProducer.

Typically you want to use a higher level API. For example the transaction API: https://github.com/apache/iceberg-python/blob/a7f207f7e5831b3be02bd023c4b33babc3ea13f6/pyiceberg/table/__init__.py#L1153-L1161

Reply:
Great! 

> Typically you want to use a higher level API. For example the transaction API: https://github.com/apache/iceberg-python/blob/a7f207f7e5831b3be02bd023c4b33babc3ea13f6/pyiceberg/table/__init__.py#L1153-L1161

That would be much cleaner but unfortunately it's not in the 0.6.0 release!
https://github.com/apache/iceberg-python/blob/cc449266e7fe0e97f23e61b3c732b75a0d0a8dec/pyiceberg/table/__init__.py#L185